src/pages/welcome/stepOne/index.js

```diff
@@ -1,38 +1,77 @@
-import ProfileIcon from "../../../assets/icons/profileIcon"
-import Form from "../../../components/form"
-import TextInput from "../../../components/form/textInput"
+import Form from "../../../components/form";
+import TextInput from "../../../components/form/textInput";
+import { useState } from "react";
 
-const StepOne = ({ data, setData }) => {
-    return (
-        <>
-            <div className="welcome-formheader">
-				<h3>Basic info</h3>
-			</div>
-            <Form className="welcome-form">
-                <div className="welcome-form-profileimg">
-                    <p className="text-blue1">Photo</p>
-                    <div className="welcome-form-profileimg-input">
-                        <ProfileIcon colour="#28C846" background="#64DC78" />
-                        <p className="text-blue1">Add headshot</p>
-                    </div>
-                    <p className="welcome-form-profileimg-error">
-                        Please upload a valid image file
-                    </p>
-                </div>
-                <div className="welcome-form-inputs">
-                    <TextInput onChange={setData} value={data.firstName} name="firstName" label={"First name"} />
-                    <TextInput onChange={setData} value={data.lastName} name="lastName" label={"Last name"} />
-                    <TextInput
-                        onChange={setData}
-                        value={data.githubUsername}
-                        name="githubUsername"
-                        label={"Github Username"}
-                    />
-                    <p className="text-blue1">*Required</p>
-                </div>
-            </Form>
-        </>
-    )
-}
+const StepOne = ({ data, setData, setValidForm }) => {
+  const [requiredFirstName, setRequiredFirstName] = useState(false);
+  const [requiredLastName, setRequiredLastName] = useState(false);
 
-export default StepOne
\ No newline at end of file
+  const checkValidFirstName = (e) => {
+    const { name, value } = e.target;
+    if (name === "firstName" && value.length !== 0) {
+      setRequiredFirstName(true);
+    } else if (name === "firstName" && value.length === 0) {
+      setRequiredFirstName(false);
+    }
+  };
+
+  const checkValidLastName = (e) => {
+    const { name, value } = e.target;
+    if (name === "lastName" && value.length !== 0) {
+      setRequiredLastName(true);
+    } else if (name === "lastName" && value.length === 0) {
+      setRequiredLastName(false);
+    }
+  };
+
+  const checkValidBothNames = () => {
+    if (requiredFirstName && requiredLastName) {
+      setValidForm(true);
+    } else {
+      setValidForm(false);
+    }
+  };
+
+  const handleInput = (e) => {
+    setData(e);
+    checkValidFirstName(e);
+    checkValidLastName(e);
+  };
+
+  checkValidBothNames();
+  console.log(requiredFirstName, requiredLastName)
```

Comment:
remove log statement

Reply:
Put it in to show the team how it was working... forgot to remove it again 🤦‍♂️ have removed it now